### DISCUSSION TITLE
Pip on iSH

### DISCUSSION THREAD

**Discussion by B1ue-Dev** [ish-app/ish#1529] [Q&A] [answered]
Is there any way to install pip on iSH? I tried to install it with "apk add python3" but even though python3 was installed, pip didn't appear.

**Comment by Eolien55** [comment, marked as answer]
You should run `python3 -m ensurepip`

It basically downloads pip and setuptools (note that wheel isn't included, so you might want to download it)

**Comment by Tommyvercetti01** [comment]
Is there any way to install pip on iSH? I tried to install it with "apk add python3" but even though python3 was installed, pip didn't appear. :(

**Comment by Xanderplayz18** [comment]
`apk add py3-pip` worked for me

**Reply by Eolien55** [reply]
Yeah, actually it's a better answer than the one I gave.

`python3 -m ensurepip` is very generic, and then the updating process is up to you. Letting it be handled by apk is just way better

**Reply by ko-kusang** [reply]
Nice

**Reply by vtoroyan** [reply]
It works for me too thanks.

**Comment by lanadelplaya** [comment]
if you are using a jailbroken iphone, you CAN actually use python -m ensurepip via mobileterminal or another terminal app (from the jailbreak repos) and install pip via the method that eolien55 mentioned, however, specifically cause it's ISH, you would need to use the apk add py3-pip command, cause it not only installs it correctly without jailbreak, but it also adds it to the path...you have to manually add the prior method mentioned above to path if you are on a jailbroken device.
